test/versioned_routing_test.rb

```diff
@@ -26,4 +26,17 @@ class VersionedRoutingTest < ActionDispatch::IntegrationTest
       get '/a_path_only_in_v3', {}, {'Accept' => 'version=2'}
     end
   end
+
+  test "a route removed in a higher version will return a 404" do
+    get '/another_path_in_v1', {}, {'Accept' => 'version=1'}
+    assert_equal(response.status, 200)
+
+    assert_raise ActionController::RoutingError do
+      get '/another_path_in_v1', {}, {'Accept' => 'version=2'}
+    end
+
+    assert_raise ActionController::RoutingError do
+      get '/another_path_in_v1', {}, {'Accept' => 'version=3'}
+    end
```

Comment:
Should these be separate tests? That would follow the pattern above and make it clearer what broken when one of these fails.